packaging/curated-cloud-installer.spec

```diff
@@ -0,0 +1,137 @@
+sed -i '/BUNDLED WITH/{N;d;}' Gemfile.lock
+
+bundle.%{ruby_version} config build.nio4r --with-cflags='%{optflags} -Wno-return-type'
+bundle.%{ruby_version} config set deployment 'true'
+bundle.%{ruby_version} config set without 'test development'
+bundle.%{ruby_version} install --local
+
+%install
+mkdir -p %{buildroot}%{app_dir}
+mkdir -p %{buildroot}%{lib_dir}
+mkdir -p %{buildroot}%{data_dir}
+
+mv vendor %{buildroot}%{lib_dir}
+mv db %{buildroot}%{data_dir}
+mv tmp %{buildroot}%{data_dir}
+mv log %{buildroot}%{data_dir}
+cp -ar . %{buildroot}%{app_dir}
+
+ln -s %{data_dir}/tmp %{buildroot}%{app_dir}/tmp
+ln -s %{data_dir}/db %{buildroot}%{app_dir}/db
+ln -s %{data_dir}/log %{buildroot}%{app_dir}/db
+
+# systemd
+mkdir -p %{buildroot}%{_unitdir}
+install -m 444 server-configs/systemd/curated-cloud-installer.service %{buildroot}%{_unitdir}
+
+# nginx
+install -D -m 644 server-configs/nginx/curated-cloud-installer.conf %{buildroot}%{_sysconfdir}/nginx/vhosts.d/curated-cloud-installer.conf
+
+# set relocated destination for gems
+sed -i -e '/BUNDLE_PATH: .*/cBUNDLE_PATH: "\/usr\/lib64\/curated-cloud-installer\/vendor\/bundle\/"' \
+    -e 's/^BUNDLE_JOBS: .*/BUNDLE_JOBS: "1"/' \
+    %{buildroot}%{app_dir}/.bundle/config
+
+# cleanup of /usr/bin/env commands
+grep -rl '\/usr\/bin\/env ruby' %{buildroot}%{lib_dir}/vendor/bundle/ruby | xargs \
+    sed -i -e 's@\/usr\/bin\/env ruby.%{ruby_version}@\/usr\/bin\/ruby\.%{ruby_version}@g' \
+    -e 's@\/usr\/bin\/env ruby@\/usr\/bin\/ruby\.%{ruby_version}@g'
+grep -rl '\/usr\/bin\/env bash' %{buildroot}%{lib_dir}/vendor/bundle/ruby | xargs \
+    sed -i -e 's@\/usr\/bin\/env bash@\/bin\/bash@g'
+
+# cleanup unneeded files
+find %{buildroot}%{lib_dir} "(" -name "*.c" -o -name "*.h" -o -name .keep ")" -delete
+find %{buildroot}%{app_dir} -name .keep -delete
+find %{buildroot}%{data_dir} -name .keep -delete
+rm %{buildroot}%{app_dir}/README.md
+rm %{buildroot}%{app_dir}/LICENSE
+rm -r  %{buildroot}%{lib_dir}/vendor/bundle/ruby/[23].*.0/cache
+rm -rf %{buildroot}%{lib_dir}/vendor/cache
+rm -rf %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/gems/*/doc
+rm -rf %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/gems/*/examples
+rm -rf %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/gems/*/samples
+rm -rf %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/gems/*/test
+rm -rf %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/gems/*/ports
+rm -rf %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/gems/*/ext
+rm -rf %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/gems/*/bin
+rm -rf %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/gems/*/spec
+rm -f %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/gems/*/.gitignore
+rm -f %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/gems/*/*/*/.gitignore
+rm -f %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/extensions/*/*/*/gem_make.out
+rm -f %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/extensions/*/*/*/mkmf.log
+rm -rf %{buildroot}%{app_dir}/server-configs
+
```

Comment:
I think that clean-up shouldn't be necessary. It should be enough to copy the files you need in the package when the `%install` section is processed, the rest will be deleted after the package has been built automatically.

Reply:
This was inherited from the RMT specfile: https://github.com/SUSE/rmt/blob/master/package/obs/rmt-server.spec#L207-L224